### PR TITLE
test: Add genesis tests

### PR DESCRIPTION
**Pull request by hallazzang**
## Description

Ref: #119 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Appropriate labels applied
- [ ] Targeted PR against correct branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: x/farming/keeper/genesis.go [hunk 1]
```diff
@@ -13,6 +14,7 @@ func (k Keeper) InitGenesis(ctx sdk.Context, genState types.GenesisState) {
 	ctx, writeCache := ctx.CacheContext()
 
 	k.SetParams(ctx, genState.Params)
+	// TODO: what if CurrentEpochDays field was empty?
 	k.SetCurrentEpochDays(ctx, genState.CurrentEpochDays)
```
[the remark this answers is not in the document]
Comment (hallazzang) [reply x/farming/keeper/genesis.go:18]: Oh, I thought that we can have zero current epoch days as valid, but I just realized that `GetCurrentEpochDays` always returns not-zero result(if the state is not set, then it returns `Params.NextEpochDays`).
Strangely we already have validation logic to check if current epoch days is zero, so current state would be just ok.

I see `AppModuleBasic` has a method `ValidateGenesis` which calls `types.ValidateGenesis` internally, then the validation logic is already being run, is it right?

[the remark this answers is not in the document]
Comment (hallazzang) [reply x/farming/keeper/genesis.go:18]: Oh, I thought that we can have zero current epoch days as valid, but I just realized that `GetCurrentEpochDays` always returns not-zero result(if the state is not set, then it returns `Params.NextEpochDays`).
Strangely we already have validation logic to check if current epoch days is zero, so current state would be just ok.

I see `AppModuleBasic` has a method `ValidateGenesis` which calls `types.ValidateGenesis` internally, then the validation logic is already being run, is it right?

[the remark this answers is not in the document]
Comment (dongsam) [reply x/farming/keeper/genesis.go:18]: Do you know where `AppModuleBasic` calls `ValidateGenesis`? 
I only found for `gentx` and `ValidateGenesisCmd` not including initGenesis phase

[the remark this answers is not in the document]
Comment (hallazzang) [reply x/farming/keeper/genesis.go:18]: Here:
https://github.com/tendermint/farming/blob/3323424035b17550fe5915c67d2f7d4b0a7221a7/x/farming/module.go#L54-L62

[the remark this answers is not in the document]
Comment (hallazzang) [reply x/farming/keeper/genesis.go:22]: Decided to call `ValidateGenesis` inside `InitGenesis`.

**Review by dongsam — APPROVED** [review]
LGTM

**Pull request merged**